bioinformatics-ua/dicoogle
# Webapp integrated with webcore! Web UI plugins at last!

This is an extended contribution to our webapp. After some tinkering, I have managed to integrate the webcore with Dicoogle. In order to achieve this, I had to modify a lot of the webapp's main core, but they were certainly worth it. Resolves #130.

- Upgraded React-router (1.0.0) and reimplemented them properly.
    - Modified routing.js into sidebar.js, which only contains the side-bar with route links.
    - Modified app.js to contain the base React element of the application, as well as all routes. All document body is now rendered by React.
- Updated `dicoogle-webcore` to 0.9.0: contains a new event 'menu' and supports specifying a plugin name in a `<dicoogle-slot>` element.
- Included dicoogle-webcore dependency in webapp, integrated.
- Used envify to define `Endpoints.base` with an environment variable. This makes it easier to define the base URL in build-time.

This had some major modifications, so it must be tested thoroughly. 

Testing:
1. Build everything as usual: build the webapp with `npm install`, then package dicoogle with `mvn install`.
2. In your Dicoogle running directory, make a "WebPlugins" directory
3. Copy or link your web UI plugin folders (try "dicoogle-webcore/test/dummy" and "dicoogle-webcore/test/react-todo") into "WebPlugins".
4. Run Dicoogle and check out the menu side-bar.
